### DISCUSSION TITLE
URGENT: Search API Not Returning Newly Created Content

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#225] [Q&A] [answered]
Hi @pallakartheekreddy and team,

We are facing a blocking issue with the Search API in our inQuiry-Knowlg setup. Everything was working fine, but for the last few days, any newly created content or question sets are not showing up in the search results.

**Error details:**
**Error in search-indexer-job-manager:** The log shows a ClassCastException due to a Boolean being cast to a String in the CompositeSearchIndexerFunction class:
```
Caused by: org.sunbird.job.exception.InvalidEventException: class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
    at org.sunbird.job.searchindexer.functions.CompositeSearchIndexerFunction.processElement(CompositeSearchIndexerFunction.scala:48) ~[search-indexer-1.0.0.jar:?]
```

**Error in search-indexer-task-manager:** Similarly, there is a ClassCastException involving a Boolean to String cast:
`java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class java.lang.String
`

I’ve attached the complete logs from the Search-Service, search-indexer-job-manager, and search-indexer-task-manager for further context.

[sunbird-search-765fb79b56-2nrjx-1729593925529524030.log](https://github.com/user-attachments/files/17474987/sunbird-search-765fb79b56-2nrjx-1729593925529524030.log)

[sunbird-search-indexer-jobmanager-69b47d56d7-2xbzr-1729593913647985201.log](https://github.com/user-attachments/files/17474996/sunbird-search-indexer-jobmanager-69b47d56d7-2xbzr-1729593913647985201.log)

[sunbird-search-indexer-taskmanager-6cb4d8ff5f-25rcm-1729593921499218387.log](https://github.com/user-attachments/files/17474999/sunbird-search-indexer-taskmanager-6cb4d8ff5f-25rcm-1729593921499218387.log)

**Request:**
This issue is blocking us, and we need help urgently. Could someone please provide insights or a solution as soon as possible?

Your quick response would be greatly appreciated!

cc - @coolbung @itsvick @snehal0904 @kk9 @suvarnakale @AbhilashKD

**Comment by pallakartheekreddy** [comment]
@rajnishdargan We are checking the logs shared by you. Will update once we find the issue

**Reply by rajnishdargan** [reply, marked as answer]
@pallakartheekreddy

The search indexer is currently up and functioning. It appears that some changes were made by our team, and we’ll be looking into this internally. If the issue reoccurs, we’ll be sure to reach out for further assistance.

Thank you for taking the time to connect over the call to discuss the issue. Closing the discussion for now.
